src/Illuminate/Http/RedirectResponse.php

```diff
@@ -82,6 +82,29 @@ public function withInput(array $input = null)
 
         return $this;
     }
+    
+    /**
+     * Append the query string parameters from the last request.
+     *
+     * @param  bool  $overwriteOld
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    function withQueryString($overwriteOld = true)
+    {
+        parse_str($_SERVER['QUERY_STRING'], $oldParams);
```

Comment:
Is this to do with the aversion of using $_SERVER variables (in case they don't exist under certain situations)?

I could do something like an isset() - but it makes the code a bit messy. I'm not sure how to achieve this elegantly.